flink-rpc/flink-rpc-core/src/test/java/org/apache/flink/runtime/rpc/RpcSystemTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.rpc;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.testutils.junit.extensions.ContextClassLoaderExtension;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import javax.annotation.Nullable;
+
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class RpcSystemTest {
+
+    @RegisterExtension
+    static final ContextClassLoaderExtension classLoaderExtension =
+            ContextClassLoaderExtension.builder()
+                    .withServiceEntry(
+                            RpcSystemLoader.class,
+                            // usually loader 2 would be loaded first
+                            TestRpcSystemLoader2.class.getName(),
+                            TestRpcSystemLoader1.class.getName())
+                    .build();
+
+    @Test
+    void testRpcSystemPrioritization() {
+        assertThat(RpcSystem.load()).isNotNull();
```

Comment:
```suggestion
        assertThat(RpcSystem.load()).as("Only " + TestRpcSystemLoader1.class.getSimpleName() + " returns a non-null value.").isNotNull();
```
nit: just to underline that `null` has some semantics here?